plugins/Live/API.php

```diff
@@ -277,6 +282,10 @@ public function getFirstVisitForVisitorId($idSite, $visitorId)
     {
         Piwik::checkUserHasSomeViewAccess();
 
+        if (!Live::isVisitorProfileEnabled($idSite)) {
+            throw new Exception('Visitor profile has been disabled in website settings');
```

Comment:
BTW maybe there could be `Live::checkVisitorProfileEnabled()` to have less duplicate code? Ideally it was to mention whether it is disabled in website settings or globally but not too important